src/services/catalog.service.ts

```diff
@@ -145,26 +145,56 @@ export class ImageItem implements IServiceItem {
   public description: string;
   public longDescription: string;
   public resource: any;
-  public isBuilder: boolean;
-  public isHidden: boolean;
+  private validSpecTagName: any;
   private tags: any;
   private catalogSrv: CatalogService;
 
   constructor (image: any, catalogSrv : CatalogService) {
     this.resource = image;
     this.catalogSrv = catalogSrv;
-    this.tags = this.catalogSrv.$filter('imageStreamTagTags')(this.resource);
-    this.iconClass = this.getIcon();
-    this.name = this.getName();
-    this.description = this.getDescription();
-    this.longDescription = this.getLongDescription();
-    this.catsBySubCats = this.getCategoriesBySubCategories();
-    this.isBuilder = _.includes(this.tags, 'builder');
-    this.isHidden = _.includes(this.tags, 'hidden');
+    this.validSpecTagName = this.getValidSpecTagName();
+    if (this.validSpecTagName) {
+      this.tags = this.getTags();
+      this.iconClass = this.getIcon();
+      this.name = this.getName();
+      this.description = this.getDescription();
+      this.longDescription = this.getLongDescription();
+      this.catsBySubCats = this.getCategoriesBySubCategories();
```

Comment:
Potentially the icon, description, and tags can be different for each spec tag. I'm not sure the ImageItem model we've made is correct.

This is OK for the catalog tile itself, but the dialog will need to do something else to support selecting versions.

I suggest when possible we work with the API objects directly. Kubernetes and OpenShift have well-defined, well-documented APIs. As it stands, we have to backtrack through the code when looking at a view to see what a property really is. And things are getting lost in translation. Here we're dropped data (icons and descriptions for other tags).